Automattic/wp-calypso
# Add Abandoned Cart Survey Hotjar trigger

#### Changes proposed in this Pull Request

This adds a `abandoned_cart_survey` Hotjar trigger which is activated when:
- the user has no previous purchases
- Hotjar is loaded (ie. analytics is allowed for the user)
- there are stale items in the cart (10 minutes)

The implementation is similar to the [current-site/stale-cart-notice](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/current-site/stale-cart-items-notice.js) but kept it separate so it can be enabled/disabled individually.

#### Testing instructions

* A Hotjar survey needs to be created with a [JavaScript trigger](https://help.hotjar.com/hc/en-us/articles/115011624487-Using-JavaScript-Triggers-for-Feedback-Tools) called `abandoned_cart_survey`. The `EN - Test Survey - use for testing only` from the wordpress.com account has been activated and configured to react to this trigger. It is also configured to always appear, for testing purposes.
* Make sure you're browsing from the US. Alternatively, you can also set the `country_code` to `US`.
* Sign up with a new user and select a Plan. You can also use a user which never purchased anything before, and add a Plan to the cart.
* Exit from the Checkout page and wait for at least 10 minutes, for the cart item to become stale.
* Refresh and the Hotjar Survey popover should appear. The Stale Cart Notice should also be visible.

![Annotation on 2021-06-09 at 13-30-22](https://user-images.githubusercontent.com/2749938/121359647-5c60ca80-c93c-11eb-8393-3bf19b4ef129.png)
